llvm/test/Transforms/GVN/matrix-intrinsics.ll

```diff
@@ -0,0 +1,85 @@
+; NOTE: Assertions have been autogenerated by utils/update_test_checks.py UTC_ARGS: --version 6
+; RUN: opt -passes=gvn -S %s | FileCheck %s
+
+define <8 x double> @redundant_unstrided_load(ptr %src) {
```

Comment:
I might have missed it, but I think it would also be good to add a few tests where we have 2 matrix stores with different number of rows and columns for DSE (i.e 2 stores to the same pointer one storing more than the other, with both possible orderings; one case should eventually be simplified). And something similar for GVN with loads